specs/data_structures.md

```diff
@@ -320,31 +286,178 @@ A proof into an SMT is structured as:
+| `transaction`  | `Transaction` | Actual transaction.                                                                               |
+| `messageStart` | `uint64`      | _Optional_. Share index (in row-major order) of first share of message this transaction pays for. |
 
-## MessageData
+### Transaction
+
+| name                    | type                      | description                                                                                                                                                                                                                                                           |
+| ----------------------- | ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| TODO                    |                           |                                                                                                                                                                                                                                                                       |
+| `messageSize`           | `uint64`                  | Size of message this transaction pays a fee for, in `byte`s. If this transaction does not pay a fee for a message, must be `0`.                                                                                                                                       |
+| `messageShareRootsRoot` | [HashDigest](#hashdigest) | Merkle root of message share roots of an optional message that this transaction pays a fee to be included in the current block. Messages are split into shares and committed to here. Large messages can span across rows, which requires more roots to the provided. |
+
+
+## IntermediateStateRootData
+
+| name                            | type                                                              | description                               |
+| ------------------------------- | ----------------------------------------------------------------- | ----------------------------------------- |
+| `wrappedIntermediateStateRoots` | [WrappedIntermediateStateRoot](#wrappedintermediatestateroot)`[]` | List of wrapped intermediate state roots. |
+
+### WrappedIntermediateStateRoot
+
+| name                    | type         | description                                                                                                     |
+| ----------------------- | ------------ | --------------------------------------------------------------------------------------------------------------- |
+| `index`                 | `uint64`     | Index of this intermediate state root in the list of intermediate state roots. This is needed for fraud proofs. |
```

Comment:
This probably will become clearer later when fraud proofs are described in detail.